catalog/core/catalog-core-validator/src/main/java/ddf/catalog/validation/impl/validator/MatchAnyValidator.java

```diff
@@ -94,4 +97,24 @@ public MatchAnyValidator(List<AttributeValidator> validators) {
     }
     return Optional.of(result);
   }
+
+  @Override
+  public int hashCode() {
+    return new HashCodeBuilder(23, 37).append(validators).toHashCode();
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+
+    if (obj == null || getClass() != obj.getClass()) {
+      return false;
+    }
+
+    MatchAnyValidator that = (MatchAnyValidator) obj;
+
+    return new EqualsBuilder().append(validators, that.validators).isEquals();
```

Comment:
❓ I don't know the details of `EqualsBuilder` so how is equality of collections of objects handled? Is it pair-wise check of values in each index? Does it perform a sort before testing? If so then what kind of sort?

Reply:
It doesn't guarantee that and a `sort()` is required before checking equality of `list` such as `validators` in this case. I added a new commit that adds sorting of validators based on hash code of each validators.